core/api/export/cp_data_extraction_all.py

```diff
@@ -0,0 +1,417 @@
+    header_row_start_idx = 1
+
+    def __init__(self, wb, year):
+        headers = [
+            {
+                "id": "country_name",
+                "headerName": "Country",
+            },
+            {
+                "id": "chemical_name",
+                "headerName": "Chemicals",
+            },
+            {
+                "id": "previous_year_price",
+                "headerName": "Previous Year Price",
+            },
+            {
+                "id": "current_year_price",
+                "headerName": str(year),
+            },
+            {
+                "id": "remarks",
+                "headerName": "Remarks",
+            },
+            {
+                "id": "notes",
+                "headerName": "Notes",
+            },
+        ]
+        sheet = wb.create_sheet("ODSPrice")
+        super().__init__(sheet, headers)
+
+    def write_data(self, data):
+        row_idx = self.header_row_end_idx + 1
+        for price in data:
+            self._write_price_row(row_idx, price)
+            row_idx += 1
+
+    def _write_price_row(self, row_idx, price):
+        for header_id, header in self.headers.items():
+            if header_id == "country_name":
+                value = price.country_programme_report.country.name
+            elif header_id == "chemical_name":
+                value = price.get_chemical_display_name()
+            else:
+                value = getattr(price, header_id, None)
+
+            self._write_record_cell(
+                row_idx,
+                header["column"],
+                value,
+            )
+
+
+class CPDetailsExtractionWriter(BaseWriter):
+    header_row_start_idx = 1
+
+    def __init__(self, wb, year):
+        headers = [
+            {
+                "id": "country_name",
+                "headerName": "Country",
+            },
+            {
+                "id": "substance_group",
+                "headerName": "Annex Group",
+            },
+            {
+                "id": "substance_name",
+                "headerName": "Substances",
+            },
+            {
+                "id": "substance_odp",
+                "headerName": "ODP Conversion for HCFC",
+            },
+            {
+                "id": "substance_gwp",
+                "headerName": "GWP for HFC",
+            },
+            {
+                "id": "record_value",
+                "headerName": str(year),
+            },
+            {
+                "id": "notes",
+                "headerName": "Notes",
+            },
+        ]
+        sheet = wb.create_sheet("CP-Details")
+        super().__init__(sheet, headers)
+
+    def write_data(self, data):
+        row_idx = self.header_row_end_idx + 1
+        for record in data:
+            self._write_record_row(row_idx, record)
+            row_idx += 1
+
+    def _write_record_row(self, row_idx, record):
+        for header_id, header in self.headers.items():
+            if header_id == "country_name":
```

Comment:
You could have also used a header_id - getter mapping like:
```
{
    "country_name": lambda record: record.country_programme_report.country.name,
    ...
}
```
but I think the current implementation is more readable/easier to maintain than my suggestion, so no need to change it.